### DISCUSSION TITLE
Do I have to restart vite everytime I update routes (file based routing)?

### DISCUSSION THREAD

**Discussion by ashoktamang** [TanStack/router#2330] [Q&A]
I am using Vite + React + Typescript. So, everytime I update routes, I have to restart vite to generate the updated routes. Is there a way to accomplish regeneration without restarting the server manually?

**Comment by ashoktamang** [comment]
Using `bun` runtime causes this issue.

**Reply by schiller-manuel** [reply]
the tanstack router vite plugin is regenerating the route tree while the dev server is running.
see https://tanstack.com/router/latest/docs/framework/react/guide/file-based-routing#configuration-with-vite

if this does not work with bun, please provide a minimal complete example.
